// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////

syntax = "proto3";

package google.crypto.tink;

import "proto/tpm_common.proto";
import "proto/hmac.proto";

option go_package = "github.com/google/tink/proto/tpm_hmac";

message TpmHmacParams {
  HmacParams hmac_params = 1;
  TpmObjectAuthPolicy auth_policy = 2;
  repeated TpmPrimaryKey parents = 3;
}

// key_type: type.googleapis.com/google.crypto.tink.TpmHmacKey
message TpmHmacKey {
  uint32 version = 1;
  TpmHmacParams params = 2;
  bytes wrapped_key = 3;
  bytes public_template = 4;
}

message TpmHmacKeyFormat {
  uint32 version = 1;
  TpmHmacParams params = 2;
}


